skiko/src/jvmMain/kotlin/org/jetbrains/skiko/SkikoProperties.kt

```diff
@@ -62,6 +62,26 @@ internal object SkikoProperties {
     }
 
     fun fallbackRenderApiQueue(initialApi: GraphicsApi) : List<GraphicsApi> {
+        when(initialApi) {
+            GraphicsApi.DIRECT3D -> {
+                when(hostOs) {
+                    OS.JS,
+                    OS.Ios,
+                    OS.Android,
+                    OS.Linux,
+                    OS.MacOS -> throw Exception("$hostOs does not support DirectX rendering API.")
+                }
+            }
+            GraphicsApi.METAL -> {
```

Comment:
macOs also doesn't support SOFTWARE_FAST, and OPENGL

Maybe it is easier to check fallbackApis?
```
val indexOfInitialApi = fallbackApis.indexOf(initialApi)
require(indexOfInitialApi >= 0) {
    "$hostOs does not support $initialApi rendering API."
}
fallbackApis = fallbackApis.drop(indexOfInitialApi + 1)
```

Reply:
Yes, good point. Done.